lib/Doctrine/ORM/Query/Parameter.php

```diff
@@ -49,6 +49,13 @@ class Parameter
      */
     private $type;
 
+    /**
+     * Whether the parameter type was explicitly specified or not
+     *
+     * @var bool
+     */
+    private $typeSpecified;
```

Comment:
I'm not 100% happy about this, but so far, the constructor of `Parameter` is destructive (this information is lost), hence this solution allows us to preserve the initial scope.

Changing the constructor is not viable, as it will lead to even more BC breaks.